tests/test_repocard_data.py

```diff
@@ -0,0 +1,197 @@
+import unittest
+
+import pytest
+
+import yaml
+from huggingface_hub.repocard_data import (
+    DatasetCardData,
+    EvalResult,
+    ModelCardData,
+    eval_results_to_model_index,
+    model_index_to_eval_results,
+)
+
+
+DUMMY_METADATA_WITH_MODEL_INDEX = """
+language: en
+license: mit
+library_name: timm
+tags:
+- pytorch
+- image-classification
+datasets:
+- beans
+metrics:
+- acc
+model-index:
+- name: my-cool-model
+  results:
+  - task:
+      type: image-classification
+    dataset:
+      type: beans
+      name: Beans
+    metrics:
+    - type: acc
+      value: 0.9
+"""
+
+
+class ModelCardDataTest(unittest.TestCase):
+    def test_eval_results_to_model_index(self):
+        expected_results = yaml.safe_load(DUMMY_METADATA_WITH_MODEL_INDEX)
+
+        eval_results = [
+            EvalResult(
+                task_type="image-classification",
+                dataset_type="beans",
+                dataset_name="Beans",
+                metric_type="acc",
+                metric_value=0.9,
+            ),
+        ]
+
+        model_index = eval_results_to_model_index("my-cool-model", eval_results)
+
+        self.assertEqual(model_index, expected_results["model-index"])
+
+    def test_model_index_to_eval_results(self):
+        model_index = [
+            {
+                "name": "my-cool-model",
+                "results": [
+                    {
+                        "task": {
+                            "type": "image-classification",
+                        },
+                        "dataset": {
+                            "type": "cats_vs_dogs",
+                            "name": "Cats vs. Dogs",
+                        },
+                        "metrics": [
+                            {
+                                "type": "acc",
+                                "value": 0.85,
+                            },
+                            {
+                                "type": "f1",
+                                "value": 0.9,
+                            },
+                        ],
+                    },
+                    {
+                        "task": {
+                            "type": "image-classification",
+                        },
+                        "dataset": {
+                            "type": "beans",
+                            "name": "Beans",
+                        },
+                        "metrics": [
+                            {
+                                "type": "acc",
+                                "value": 0.9,
+                            }
+                        ],
+                    },
+                ],
+            }
+        ]
+        model_name, eval_results = model_index_to_eval_results(model_index)
+
+        self.assertEqual(len(eval_results), 3)
```

Comment:
hmmm I Might be missing something but aren't there only 2 eval results above?

Reply:
Ah we have two metrics in the first result. Shouldn't there be 2 metrics in the same result or do we split into multiple results?

Reply:
There are 3 eval results:
- 2 for the cats-vs-dogs dataset and 
- one for the beans dataset. 

The `EvalResult` object is a flattened representation, so each instance represents one evaluation result coming from one dataset.

Reply:
So, the above would be something like this:

```python
from huggingface_hub import EvalResult

eval_results = [
    EvalResult(dataset_type='cats_vs_dogs', metric_type='acc', metric_value=0.85, ...),
    EvalResult(dataset_type='cats_vs_dogs', metric_type='f1', metric_value=0.9, ...),
    EvalResult(dataset_type='beans', metric_type='acc', metric_value=0.9, ...)
]
```

Reply:
Makes sense, thanks for explaining!